src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildAnAppWithoutTransitiveProjectRefs.cs

```diff
@@ -98,18 +177,159 @@ public void It_builds_the_project_successfully_when_RAR_does_not_find_all_refere
+                ReferencedProjects = { project3 },
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "2", string.Empty)
+                }
+            };
+            
+            var project1 = new TestProject
+            {
+                Name = "1",
+                IsExe = true,
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.1",
+                ReferencedProjects = { project2 },
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "1", string.Empty)
+                }
+            };
+
+            return project1;
+        }
+
+        private TestProject DiamondShapeGraphWithRuntimeDependencies()
+        {
+            var project5 = new TestProject
+            {
+                Name = "5",
+                IsSdkProject = true,
+                TargetFrameworks = "netstandard1.3",
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "5", string.Empty)
+                }
+            };
+            
+            var project4 = new TestProject
+            {
+                Name = "4",
+                IsSdkProject = true,
+                TargetFrameworks = "netstandard1.3;netstandard1.6;net461",
+                ReferencedProjects = { project5 },
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "4", "_5.Class1.Message();")
+                }
+            };
+            
+            var project3 = new TestProject
+            {
+                Name = "3",
+                IsSdkProject = true,
+                TargetFrameworks = "netstandard2.0;net462",
+                ReferencedProjects = { project4 },
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "3", "_4.Class1.Message();")
+                }
+            };
+            
+            var project2 = new TestProject
+            {
+                Name = "2",
+                IsSdkProject = true,
+                TargetFrameworks = "netstandard1.5",
+                ReferencedProjects = { project4 },
+                SourceFiles = new Dictionary<string, string>
+                {
+                    ["Program.cs"] = string.Format(SourceFile, "2", "_4.Class1.Message();")
+                }
+            };
+            
+            var project1 = new TestProject
+            {
+                Name = "1",
+                IsExe = true,
+                IsSdkProject = true,
+                TargetFrameworks = "netcoreapp2.1;net472",
+                ReferencedProjects = { project2, project3 },
+                SourceFiles = new Dictionary<string, string>
```

Comment:
```suggestion
                SourceFiles =
```

I think you don't need the setter in that case.

Reply:
True, I could mutate the default empty dictionary, but then I can't nicely initialize everything in the initialization block :)

Reply:
The initialization block will work:

```
Microsoft (R) Visual C# Interactive Compiler version 3.2.0-beta4-19321-04 ()
Copyright (C) Microsoft Corporation. All rights reserved.

Type "#help" for more information.
> using System.Collections.Generic;
> class C { public Dictionary<string, string> D { get; } = new Dictionary<string, string>(); }
> var c = new C { D = { ["A"] = "B", ["C"] = "D" } };
> c.D["A"]
"B"
> c.D["C"]
"D"
```

Reply:
It will call the indexer setter:

```
 IL_0000:  newobj     instance void C::.ctor()
  IL_0005:  dup
  IL_0006:  callvirt   instance class [mscorlib]System.Collections.Generic.Dictionary`2<string,string> C::get_D()
  IL_000b:  ldstr      "A"
  IL_0010:  ldstr      "B"
  IL_0015:  callvirt   instance void class [mscorlib]System.Collections.Generic.Dictionary`2<string,string>::set_Item(!0, !1)
  IL_001a:  dup
  IL_001b:  callvirt   instance class [mscorlib]System.Collections.Generic.Dictionary`2<string,string> C::get_D()
  IL_0020:  ldstr      "C"
  IL_0025:  ldstr      "D"
  IL_002a:  callvirt   instance void class [mscorlib]System.Collections.Generic.Dictionary`2<string,string>::set_Item(!0, !1)
```

Reply:
Learning something new every time :)